### PR TITLE
Fix(SCT-680, SCT-728): Fix date bug when ending a warning note

### PR DESCRIPTION
**Pull request by Renny4Real**
There is a potential issue with the date being provided by the client app when ending a warning note. 

Ending a warning note currently relies on the device time being passed into the request, but there may be instances where this might be different to our server time. e.g. if the client device time is incorrect or possible timezone differences.

This checks for the `status` parameter in an incoming PATCH warning note request and saves the server's current time to a warning note's `end date` if the status is set as `closed`. Thus, ignoring any end date value passed in from the frontend.

A [future PR](https://hackney.atlassian.net/browse/SCT-727) would remove the `ended date` parameter from being passed in Frontend.

### CODE REVIEW

**Review by neilkidd — COMMENTED** [review]
Hiya,

Do we know what the business rule is for ending warning notes?
Is there a good reason the api should permit a date to be passed in? (This seems to open the system up to the issues we are now trying to solve)

**Comment by martapederiva** [comment]
> Is there a good reason the api should permit a date to be passed in? (This seems to open the system up to the issues we are now trying to solve)

As far as I understand, the date needs to be saved in the DB for auditing reasons. Whether the date itself needs to be subjected to a test, given that it's not user input, was another point raised.

**Review by martapederiva — APPROVED** [review]
Looks good to me 🙂

**Comment by chrishutchinson** [comment]
@Renny4Real my general sense is that the front end _shouldn't_ even pass an end date. Instead, if the the front end passes a `status` of `"closed"` (which I believe indicates the user wants to "end" the Warning Note) then the API should then generate an end date/timestamp at that point, and persist it to the database along with the rest of the review data.

This eliminates any validation need on this field as the API will handle all of it and we'll guarantee that the date/time is always correct (according to the backend server).

**Comment by Renny4Real** [comment]
> @Renny4Real my general sense is that the front end _shouldn't_ even pass an end date. Instead, if the the front end passes a `status` of `"closed"` (which I believe indicates the user wants to "end" the Warning Note) then the API should then generate an end date/timestamp at that point, and persist it to the database along with the rest of the review data.
> 
> This eliminates any validation need on this field as the API will handle all of it and we'll guarantee that the date/time is always correct (according to the backend server).

@chrishutchinson Yeah, that makes sense. 
Should we be considering the possibility of custom end dates at this point or maybe when it comes up & discuss it then? 
I don't think it blocks the approach you suggested but wondering if we should be thinking about it as @neilkidd makes a good point around what the business rule is for ending a warning note.

File: SocialCareCaseViewerApi.Tests/V1/Boundary/Request/PatchWarningNoteRequestTests.cs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        public void GivenARequestWithReviewNotesMoreThanMaximumLengthValidationFails()
+        {
+            var request = new PatchWarningNoteRequest
+            {
+                ReviewNotes = new string('a', 1100),
+                Status = "open"
+            };
+
+            _classUnderTest.ShouldHaveValidationErrorFor(x => x.ReviewNotes, request)
+                .WithErrorMessage("Review notes be less than 1000 characters");
```
[the remark this answers is not in the document]
Comment (Renny4Real) [reply SocialCareCaseViewerApi.Tests/V1/Boundary/Request/PatchWarningNoteRequestTests.cs]: Done in aeac39a

**Review by neilkidd — APPROVED** [review]
LGTM

**Pull request merged**